pkg/client/client.go

```diff
@@ -761,6 +763,126 @@ func (c *Client) WaitForPrometheus(ctx context.Context, p *monv1.Prometheus) err
 	return nil
 }
 
+// ValidationError indicates if an operation performed by the client has
+// resulted in an invalid state.
+type ValidationError struct {
+	Degraded    bool
+	Unavailable bool
+	Reasons     []string
+}
+
+func (ve *ValidationError) Error() string {
+	return fmt.Sprintf("error: %s degraded: %t unavailable: %t",
+		strings.Join(ve.Reasons, ";"), ve.Degraded, ve.Unavailable)
+}
+
+var _ error = (*ValidationError)(nil)
+
+func (c *Client) validatePromethuesPodState(ctx context.Context, p *monv1.Prometheus) *ValidationError {
+
+	status, _, err := prometheusoperator.Status(ctx, c.kclient.(*kubernetes.Clientset), p)
+	if err != nil {
+		klog.V(4).ErrorS(err, "validatePrometheusPodState: failed to get Prometheus status")
+		// TODO(sthaha): do we want to not set Degraded in this case?
```

Comment:
@simonpasquier @jan--f Any thoughts if we want to set Degraded if prometheusoperator.Status returns err?

Reply:
`Unknown` might be the best choice?

Reply:
@simonpasquier do you mean we  add an `Unknown` as well to ValidationError?